src/hotspot/os/aix/loadlib_aix.cpp

```diff
@@ -223,7 +242,12 @@ static bool reload_table() {
     lm->data     = ldi->ldinfo_dataorg;
     lm->data_len = ldi->ldinfo_datasize;
 
-    lm->path = g_stringlist.add(ldi->ldinfo_filename);
+    if (pgmbase != nullptr && 0 == strcmp(pgmbase, ldi->ldinfo_filename)) {
```

Comment:
Maybe better `strcmp(...) == 0 ` this is what we use almost everywhere in the HS codebase .